app/code/Magento/Sales/Model/Order/ShippingMethod.php

```diff
@@ -0,0 +1,332 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Sales\Model\Order;
+
+/**
+ * Value object for shipping_method order attribute
+ */
+class ShippingMethod extends \Magento\Framework\DataObject
+{
+    /**
+     * @var
+     */
+    private $carrierCode;
+    /**
+     * @var
+     */
+    private $method;
+
+    public function __construct(string $carrierCode, string $method)
+    {
+        $this->carrierCode = $carrierCode;
+        $this->method = $method;
+    }
+
+    /**
+     * Shipping method as in shipping_method order attribute
+     *
+     * @param string $fullShippingMethod
+     * @return ShippingMethod
+     * @throws \InvalidArgumentException
+     */
+    public static function fromFullShippingMethodCode(string $fullShippingMethod) : ShippingMethod
```

Comment:
I'm afraid I must insist. Introduction of new static methods is forbidden and will cause failure of internal tests.

BTW Magento also does not allow to use `new` operator to instantiate objects (except exceptions and embedded classes such as `DateTime`) and this static method is hiding such implementation (in case of a factory it should use object manager to create an instance of a class).

Another one strong reason from my point of view is that knowledge about full shipping method code format now is spread across several classes (this is bad practice and lead to tight coupling). And it is better to avoid infection of this new class by the legacy implementation. Especially in class signature. "Full code" is a foreign concept for Shipping Method. And is actually a bridge from legacy implementation to a new data structure. Thereby, it should be extracted in a separate class.